decidim-budgets/app/models/decidim/budgets/order.rb

```diff
@@ -16,9 +16,17 @@ class Order < Budgets::ApplicationRecord
       validates :user, presence: true, uniqueness: { scope: :feature }
       validate :user_belongs_to_organization
 
+      validates(:total_budget, numericality: {
+                  greater_than_or_equal_to: :minimum_budget,
+                }, if: :checked_out?)
+
+      validates(:total_budget, numericality: {
+                  less_than_or_equal_to: :maximum_budget
+                })
+
       # Public: Returns the sum of project budgets
       def total_budget
-        @total_budget ||= projects.sum(&:budget)
+        projects.to_a.sum(&:budget)
```

Comment:
If we do it that way it doesn't work when projects aren't stired, which is exactly what we wanna do